### ISSUE TITLE
Constructed element with name document is flagged as error

### ISSUE THREAD

**Issue by sanderson-sfdc** [overstory/marklogic-intellij-plugin#56]
In the following code:

`declare function local:documentAsNCName() {
    let $example :=
        element document {
            "text in a an element named document"
        }
    return $example
};`

the plugin flags the element name document as an error, saying:

> `(`, NCName, URIQualifiedName or `{` expected, got document

In this case, though, document is an NCName.  

Document is not a reserved function name (see https://www.w3.org/TR/2017/REC-xquery-31-20170321/#id-reserved-fn-names) and is an allowed NCName here.

**Comment by ronhitchens** [comment]
You are correct that "document" is not a reserved word in XQuery (very few are), but the XQuery parser used by this plugin is not 100% correct.  There are a few corner cases that are not handled correctly, including this one.  The parser comes from the project from which this project was forked (https://github.com/ligasgr/intellij-xquery).  Making corrections to the grammar requires more than a simple rebuild.  I have accumulated a few grammar corrections that need to be done and will address them in the future.  My personal workload prevents me from devoting very much time to this project right now.

The code example you've given will run ok on MarkLogic but will show as an error in the editor, as you describe.

However, there is a workaround for this issue.  If you use the more verbose element constructor syntax then the plugin's parser will be happy.  As defined here: https://www.w3.org/TR/2017/REC-xquery-31-20170321/#id-computedElements  You can give the QName of the element to construct either as a simple inline name (which you're doing) or as a constructed QName.  The parser as it's currently implemented flags "document" as a token that is not allowed there (incorrectly).

But you can specify the element name with a constructed QName inside curly braces, like this: `xs:QName("document")`.

The code below will not show an error in the plugin and will run identically to your example.

```
declare function local:documentAsNCName() {
	let $example := element { xs:QName ("document") } {"text in an element named document"}
	return $example
};

local:documentAsNCName()
```
You can also do a shorthand QName constructor, like this:

```
declare function local:documentAsNCName() {
	let $example := element { "document" } {"text in an element named document"}
	return $example
};

local:documentAsNCName()
```

No fix is likely to be made for this soon, but it will be added to the list of parser fixes to make.
